### PR TITLE
langchain-aws InMemoryVectorStore documentation updates

### PR DESCRIPTION
**Pull request by LakshmiPeri23**
Thank you for contributing to LangChain!

- [x] **PR title**: "Add documentaiton on InMemoryVectorStore driver for MemoryDB to langchain-aws"
  - Langchain-aws repo :Add MemoryDB documentation 
  - Example: "community: add foobar LLM"

- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** Added documentation on InMemoryVectorStore driver to aws.mdx and usage example on MemoryDB clusuter
    - **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!

- [x] **Add tests and docs**: If you're adding a new integration, please include
Add memorydb notebook to docs/docs/integrations/ folde

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: docs/docs/integrations/platforms/aws.mdx [hunk 1]
```diff
@@ -197,7 +197,26 @@ See a [usage example](/docs/integrations/vectorstores/documentdb).
 ```python
 from langchain.vectorstores import DocumentDBVectorSearch
 ```
+### Amazon MemoryDB 
+[Amazon MemoryDB](https://aws.amazon.com/memorydb/) MemoryDB is a durable, in-memory database service that delivers ultra-fast performance. MemoryDB is compatible with Redis OSS, a popular open source data store,
```
Comment (3coins) [inline docs/docs/integrations/platforms/aws.mdx:201]: Amazon Memory DB and MemoryDB seems duplicate.
```suggestion
[Amazon MemoryDB](https://aws.amazon.com/memorydb/) is a durable, in-memory database service that delivers ultra-fast performance. MemoryDB is compatible with Redis OSS, a popular open source data store, 
```

File: docs/docs/integrations/platforms/aws.mdx [hunk 2]
```diff
@@ -197,7 +197,26 @@ See a [usage example](/docs/integrations/vectorstores/documentdb).
 ```python
 from langchain.vectorstores import DocumentDBVectorSearch
 ```
+### Amazon MemoryDB 
+[Amazon MemoryDB](https://aws.amazon.com/memorydb/) MemoryDB is a durable, in-memory database service that delivers ultra-fast performance. MemoryDB is compatible with Redis OSS, a popular open source data store, 
+enabling you to quickly build applications using the same flexible and friendly Redis OSS APIs, and commands that they already use today. 
 
+AWS offers services for computing, databases, storage, analytics, and other functionality. For an overview of all AWS services, see [Cloud Computing with Amazon Web Services](https://aws.amazon.com/what-is-aws/).
+
```
Comment (3coins) [inline docs/docs/integrations/platforms/aws.mdx:205]: Is this necessary? Better to be precise about the relevant service, than promoting AWS.

File: docs/docs/integrations/platforms/aws.mdx [hunk 3]
```diff
@@ -197,7 +197,26 @@ See a [usage example](/docs/integrations/vectorstores/documentdb).
...
+See a [usage example](/docs/integrations/vectorstores/memorydb).
```
Comment (3coins) [inline docs/docs/integrations/platforms/aws.mdx:206]: Should we move this to the bottom after the python code section?

File: docs/docs/integrations/platforms/aws.mdx [hunk 4]
```diff
@@ -197,7 +197,26 @@ See a [usage example](/docs/integrations/vectorstores/documentdb).
...
+InMemoryVectorStore class provides a vectorstore to connect with Amazon MemoryDB.
```
Comment (3coins) [inline docs/docs/integrations/platforms/aws.mdx:207]:
```suggestion
InMemoryVectorStore class provides a vector store to connect with Amazon MemoryDB.
```

File: docs/docs/integrations/platforms/aws.mdx (hunk 3 above)
Comment (LakshmiPeri23) [reply docs/docs/integrations/platforms/aws.mdx]: modified

File: docs/docs/integrations/vectorstores/memorydb.ipynb [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = getpass.getpass(\"OpenAI API Key:\")"
+   ]
+  },
```
Comment (3coins) [inline docs/docs/integrations/vectorstores/memorydb.ipynb:67]: This seems irrelevant.

File: docs/docs/integrations/vectorstores/memorydb.ipynb [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%pip install --upgrade --quiet  redis redisvl langchain-openai tiktoken"
```
Comment (3coins) [inline docs/docs/integrations/vectorstores/memorydb.ipynb:46]:
```suggestion
    "%pip install --upgrade --quiet  redis redisvl langchain-aws"
```

**Review by LakshmiPeri23 — COMMENTED** [review]
updated to  use langchain-aws

**Review by 3coins — APPROVED** [review]
Looks good!

**Review by ccurme — APPROVED** [review]
